app/js/sumo_db.js

```diff
@@ -12,6 +12,26 @@
   var sequence_id = 0;
   var last_request;
 
+  function show_generic_error() {
```

Comment:
I'd like to move this into another file to make sure sumo_db only does HTTP requests and no UI. Can you create an event and dispatch it ([docs for modern way to that](https://developer.mozilla.org/en-US/docs/Web/API/Document/createEvent))?
And then listen for this event in a new file. I suggest calling this event network-error.